src/librustc/middle/liveness.rs

```diff
@@ -1209,8 +1209,7 @@ impl<'a, 'tcx> Liveness<'a, 'tcx> {
           }
 
           hir::ExprKind::Call(ref f, ref args) => {
-            // FIXME(canndrew): This is_never should really be an is_uninhabited
-            let succ = if self.tables.expr_ty(expr).is_never() {
+            let succ = if self.tables.expr_ty(expr).conservative_is_uninhabited(self.ir.tcx) {
```

Comment:
I see, so presumably this is what is triggering the lint, because `TcpStream(a)` returned an uninhabited value?

Reply:
Yes, that's right.